src/number/round.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Computes `number` rounded to `precision`.
+ *
+ * @param {number} number The number to round.
+ * @param {number} [precision=0] The precision to round to.
+ * @returns {number} Returns the rounded number.
+ *
+ *
+ * round(123.456)
+ * // => 123.5
+ *
+ * round(1234.56, -2);
+ * // => 1200
+ *
+ *
+ */
+export function round(value: number, precision?: number): number {
+  precision =
+    precision == null
+      ? 0
+      : precision >= 0
+        ? Math.min(precision, 307)
+        : Math.max(precision, -323)
```

Comment:
How can we verify these precision limits to be correct?

Interestingly, Lodash does this:
```ts
precision = precision == null ? 0 : (precision >= 0 ? Math.min(precision, 292) : Math.max(precision, -292))
```

Reply:
@aleclarson  I added that check inspired from them. 
![Screenshot 2024-06-30 at 10 34 34 PM](https://github.com/radashi-org/radashi/assets/42430941/19126ca9-9a9f-4f2d-b445-8c11124f1fe3)

Reply:
Also i found an issue related with this: https://github.com/lodash/lodash/issues/4232

Reply:
Here's what Claude Sonnet 3.5 says on the matter:

> You're right to question this. Lodash's choice of 292 is indeed intentional and has a specific reason. Let me explain:
> 
> 1. Precision: While JavaScript can theoretically represent numbers up to 308 decimal places, it doesn't maintain full precision for all of those digits.
> 
> 2. Reliable accuracy: Lodash uses 292 as a more conservative limit to ensure reliable and consistent results across different JavaScript engines and environments.
> 
> 3. IEEE 754 standard: This limit is related to the IEEE 754 double-precision floating-point format, which guarantees about 15-17 decimal digits of precision.
> 
> 4. Safety margin: 292 provides a safety margin, avoiding potential rounding errors or inconsistencies that might occur when pushing to the absolute limits of JavaScript's number representation.
> 
> So, while 308 is technically the maximum exponent, Lodash's choice of 292 is a pragmatic one, prioritizing consistent and reliable results over the theoretical maximum. For most practical applications, this limit is more than sufficient and helps prevent subtle bugs that could arise from pushing the boundaries of floating-point precision.

I will push an update, along with the Lodash test suite and a reimplementation to support the edge cases that `lodash.round` does.

Reply:
Thanks @aleclarson  for the explanation. So they chose 292 to ensure results are reliable and consistent across different JavaScript engines (e.g., V8 in Chrome, SpiderMonkey in Firefox) and environments (e.g., browsers, Node.js).

Reply:
I tracked down the original commit:
68ae6fc2122c4ea23ae42f9b176e20cffde2d7ba

Not much justification is given. If we can use technically “more correct” limits, we should as long as `NaN` is avoided. If someone reports misbehavior in some JS engine, we'll consider a breaking change then, but considering such large precision values are probably very rare in Javascript code, that won't happen anytime soon.

Reply:
So to be on the safe side shall we stick with 292 then?

Reply:
After some testing, I found that `292` is the upper limit that prevents NaN in all cases, while `-323` works fine as a lower limit in all cases.